docker-app/qfieldcloud/core/drf_utils.py

```diff
@@ -0,0 +1,104 @@
+from rest_framework import filters, viewsets
+
+
+class QfcOrderingFilter(filters.OrderingFilter):
+    """
+    Custom QFC OrderingFilter class that allows to use custom expression
+    Use it in a ModelViewSet by setting the `filter_backends` and `ordering_fields` fields
+    It is possible to use an `ordering_fields` expression value with attributes
+    e.g.: "my_field::alias=my_field_alias,key=value"
+    """
+
+    SEPARATOR = "::"
+    TOKENS_LIST_SEPARATOR = ","
+    TOKENS_VALUE_SEPARATOR = "="
+
+    def _get_query_field(self, fields: list[str], term: str) -> str | None:
+        """
+        Returns a term from a field list, if present
+        """
+        for field in fields:
+            compare_value = field
+
+            if field.startswith("-"):
+                compare_value = field[1:]
+
+            if compare_value != term:
+                continue
+
+            return field
+        return None
+
+    def _get_tokenized_attributes(self, raw: str) -> dict[str, str]:
+        """
+        Parses an ordering field attributes expression
+        e.g.: "alias=my_field_alias,key=value"
+        Returns a dict containing the expression's attributes
+        """
+        definition_attrs = raw.split(self.TOKENS_LIST_SEPARATOR)
+
+        attr_dict = {}
+        for attr in definition_attrs:
+            token, value = attr.split(self.TOKENS_VALUE_SEPARATOR, 1)
+            attr_dict[token] = value
+
+        return attr_dict
+
+    def _parse_definition(self, definition: str) -> tuple[str, dict[str, str]]:
+        """
+        Parses a custom ordering field with attributes expression
+        e.g.: "my_field::alias=my_field_alias,key=value"
+        Returns a tuple containing the field definition and its attributes dict
+        """
+        name, attr_str = definition.split(self.SEPARATOR, 1)
+        attrs = self._get_tokenized_attributes(attr_str)
+
+        return name, attrs
+
+    def remove_invalid_fields(
+        self,
+        queryset: QuerySet,
+        fields: list[str],
+        view: viewsets.ModelViewSet,
+        request: HttpRequest,
+    ) -> list[str]:
+        """
+        Process ordering fields by parsing custom field expression
+        e.g.: "my_field::alias=my_field_alias,key=value"
+        In the above example, `alias` is the URL GET param value
+        but `my_field` is the real model field
+        Returns parsed ordering fields where aliases have been replaced
+        """
+        base_fields = super().remove_invalid_fields(queryset, fields, view, request)
+        valid_fields = []
+
+        for field_name, _verbose_name in self.get_valid_fields(
+            queryset, view, context={"request": request}
+        ):
+            # standard handling of fields from the base class
+            query_field_name = self._get_query_field(base_fields, field_name)
+
+            if query_field_name:
+                valid_fields.append(query_field_name)
+                continue
```

Comment:
Hi @suricactus , added this part after our session, after having noted that sorting by `-name` or `-created_at` was not sorting as expected